### PR TITLE
feat(cli): Support validator check-avs command

### LINKED ISSUE #3976: Add AVS status check
Our goal is to automatically know:
1. Which validators have enrolled in our AVS
2. Are they actually live and posting data publicly
3. Which chains are they validating for

Add Kunal's https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/3923 into Mo's pre-flight validator check PR https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/3813

### PR DESCRIPTION
**Pull request by Mo-Hussain**
### Description

- Add a command to show all the operators on a specific chain and to show all the other chains that the operator is validating on, including if they are signing the latest checkpoint
- Support an operator to check for there own validators

example usage: 

```
$ yarn hyperlan avs check--chain holesky

❗️ MerkleTreeHook is not deployed on anvil8545


Validator: 0x03c842db86A6A3E524D4a6615390c1Ea8E2b9541
Operator address: 0xbA78Cdc5B9D0742fAf9bCEdBf8336a5e96f6F9c8
  Validating on...
  ethereum
  Storage location: s3://hyperlane-mainnet3-ethereum-validator-0/us-east-1
  Latest merkle tree checkpoint index: 6888
  Latest validator checkpoint index: 6888
  ✅ Validator is signing latest checkpoint


Validator: 0x7Ab28AD88bB45867137Ea823af88E2cb02359C03
Operator address: 0xf7C9D9b4429109340883c8849893b4Fa3f3724c3
Validator is not validating on any chain


Validator: 0x2233a5ce12f814bd64c9CDd73410BB8693124d40
Operator address: 0x77A9b07Bf5C752Aa9dcFf743516D24916CE3E277
  Validating on...
  alfajores
  Storage location: s3://hyperlane-testnet4-alfajores-validator-0/us-east-1
  Latest merkle tree checkpoint index: 1952
  Latest validator checkpoint index: 1952
  ✅ Validator is signing latest checkpoint


Validator: 0x4a3D6E0Ff640f2F0C1091878E6eD3bE7ED3F6665
Operator address: 0x4a3D6E0Ff640f2F0C1091878E6eD3bE7ED3F6665
Validator is not validating on any chain


Validator: 0x8EF2BEc25b963899225f367747372Dddc7A55808
Operator address: 0x2385bf3AB1e88e2055368A73A4943faff99A644D
Validator is not validating on any chain


Validator: 0xD2cEd7fb01bF7417dA8784ab77B86AF42dA07404
Operator address: 0x2385bf3AB1e88e2055368A73A4943faff99A644D
  Validating on...
  holesky
  Storage location: s3://hyperlane-hashkey-cloud-validator/us-west-1
  Latest merkle tree checkpoint index: 21
  Latest validator checkpoint index: 21
  ✅ Validator is signing latest checkpoint


Validator: 0xe7dA95EA4fa4391dCAa3aCaB5B8a97b9DCdDABe9
Operator address: 0x4a3D6E0Ff640f2F0C1091878E6eD3bE7ED3F6665
  Validating on...
  holesky
  Storage location: s3://hyperlane-validator-signatures-ethereum/eu-north-1
  Latest merkle tree checkpoint index: 21
  Latest validator checkpoint index: 21
  ✅ Validator is signing latest checkpoint


Validator: 0xcd9403306F4E752Ee9c8cd1F4199CAe6f8BFBeFE
Operator address: 0x6C76A22052d5e8bbad9801bF1E413477843aF698
  Validating on...
  holesky
  Storage location: file:///tmp/validator-signatures
  Latest merkle tree checkpoint index: 21
  ❌ Failed to fetch latest signed checkpoint index
  The following warnings were encountered:
   ❗️ Failed to fetch latest signed checkpoint index of validator on holesky, this is likely due to failing to read an S3 bucket

```

### Related issues

- Fixes #3976 

### Backward compatibility

Yes
### Testing

Manual

### CODE REVIEW

File: typescript/cli/src/validator/check-avs.ts [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+
+        if (!latestMerkleTreeCheckpointIndex) {
+          warnings.push(
+            `❗️ Failed to fetch latest checkpoint index of merkleTreeHook on ${chain}.`,
+          );
+        }
+
+        if (!latestValidatorCheckpointIndex) {
+          warnings.push(
+            `❗️ Failed to fetch latest signed checkpoint index of on ${chain}.`,
```
Comment (aroralanuk) [inline typescript/cli/src/validator/check-avs.ts:101]:
```suggestion
            `❗️ Failed to fetch latest signed checkpoint index of validator on ${chain}.`
```

**Review by aroralanuk — COMMENTED** [review]
It will be consistent to have it as a subcommand below avs like:

```hyperlane avs checkStatus --chain holesky```

**Comment by aroralanuk** [comment]
Also, I was thinking it'd be nice for the operator to just view their own, so, if they pass in:

```hyperlane avs checkStatus --operatorKeyPath <LOCAL_KEY_FILE> --chain holesky```

they can see status for just their operating key. I can add this to the guide too for them to confirm.

(You can look at stakeRegistry for reading an encrypted file locally.)

File: typescript/cli/src/logger.ts [hunk 2]
```diff
@@ -51,3 +56,6 @@ export const errorRed = (...args: any) => logColor('error', chalk.red, ...args);
 
 // No support for table in pino so print directly to console
 export const logTable = (...args: any) => console.table(...args);
+
+export const indent = (msg: string, indentLevel = 2) =>
```
Comment (Mo-Hussain) [inline typescript/cli/src/logger.ts:60]: This could probably live somewhere else that is more appropriate

File: typescript/cli/src/avs/check.ts [hunk 3]
```diff
@@ -0,0 +1,327 @@
...
+
+const getEcdsaStakeRegistryAddress = (
+  chain: string,
+  topLevelErrors: string[],
+): Address | undefined => {
+  try {
+    return avsAddresses[chain]['ecdsaStakeRegistry'];
+  } catch (err) {
+    topLevelErrors.push(
+      `❗️ EcdsaStakeRegistry address not found for ${chain}`,
```
Comment (Mo-Hussain) [inline typescript/cli/src/avs/check.ts:323]: Feel free to suggest better copy here @aroralanuk

File: typescript/cli/src/commands/avs.ts [hunk 4]
```diff
@@ -87,3 +86,37 @@ const deregisterCommand: CommandModuleWithWriteContext<{
     process.exit(0);
   },
 };
+
+const checkCommand: CommandModuleWithWriteContext<{
+  chain: ChainName;
+  operatorKeyPath?: string;
+}> = {
+  command: 'check',
+  describe: 'Check AVS',
```
Comment (Mo-Hussain) [inline typescript/cli/src/commands/avs.ts:95]: Feel free to suggest better copy here @aroralanuk

File: typescript/cli/src/commands/avs.ts [hunk 5]
```diff
@@ -87,3 +86,37 @@ const deregisterCommand: CommandModuleWithWriteContext<{
...
+      errorRed(
+        `❌ No metadata found for ${chain}. Ensure it is included in your configured registry.`,
+      );
+      process.exit(1);
+    }
+
+    const chainMetadata = multiProvider.getChainMetadata(chain);
+
+    if (chainMetadata.protocol !== ProtocolType.Ethereum) {
+      errorRed(`\n❌ Validator AVS check only supports EVM chains. Exiting.`);
```
Comment (Mo-Hussain) [inline typescript/cli/src/commands/avs.ts:114]: Feel free to suggest better copy here @aroralanuk

File: typescript/cli/src/logger.ts (hunk 2 above)
Comment (nbayindirli) [reply typescript/cli/src/logger.ts:60]: please use this instead (also accounts for all lines and not just the first):
https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/e38d31685eef7b5b53e2c17879bd3306a34b1443/typescript/cli/src/utils/files.ts#L214

File: typescript/cli/src/avs/check.ts [hunk 6]
```diff
@@ -0,0 +1,326 @@
...
+  operatorAddress: Address;
+  chains: ChainMap<ChainInfo>;
+}
+
+export const checkValidatorAVSSetup = async (
+  chain: string,
+  context: CommandContext,
+  operatorKeyPath?: string,
+) => {
+  logBlue(`Checking AVS setup for ${chain}...`);
```
Comment (Mo-Hussain) [inline typescript/cli/src/avs/check.ts:49]: @aroralanuk - what would you suggest as a better message?

File: typescript/cli/src/avs/check.ts [hunk 7]
```diff
@@ -0,0 +1,326 @@
...
+) => {
+  const { multiProvider, registry } = context;
+  const failedToReadChains: string[] = [];
+
+  const validatorAddresses = Object.keys(avsOperatorRecord);
+
+  const chains = await registry.getChains();
+  const addresses = await registry.getAddresses();
+
+  for (const chain of chains) {
```
Comment (Mo-Hussain) [inline typescript/cli/src/avs/check.ts:158]: @aroralanuk - do we only care about EVM chains?

**Review by aroralanuk — COMMENTED** [review]
I'll take another look when the RPC failures are sorted.

**Review by aroralanuk — APPROVED** [review]
yarn test is failing but otherwise LGTM

**Pull request closed without merging**